types/rdf-js/index.d.ts

```diff
@@ -476,3 +476,179 @@ export interface DatasetCoreFactory<Q extends BaseQuad = Quad> {
      */
     dataset(quads?: Q[]): DatasetCore<Q>;
 }
+
+export interface Dataset<Q extends BaseQuad = Quad> extends DatasetCore<Q> {
+    /**
+     * Imports the quads into this dataset.
+     *
+     * This method differs from `Dataset.union` in that it adds all `quads` to the current instance, rather than
+     * combining `quads` and the current instance to create a new instance.
+     */
+    addAll(quads: Dataset<Q>|Iterator<Q>): this;
```

Comment:
Need to check if 'sequence' in the spec refers to a plain array.

Reply:
Sequence should be an array I think, as it refers to .pop(), .splice(), .shift().